### PR TITLE
customPush

### PR DESCRIPTION
**Pull request by shayb**
custom push and alert on process of doing backup

### CODE REVIEW

File: app/src/main/java/org/kinecosystem/kinit/view/backup/BackupWalletActivity.kt [hunk 1]
```diff
@@ -10,11 +10,22 @@ import org.kinecosystem.kinit.viewmodel.backup.BackupModel
 class BackupWalletActivity : SingleFragmentActivity(), BackupActions {
 
     private var model: BackupModel = BackupModel()
+    private var resendEmailConut = 0
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/view/backup/BackupWalletActivity.kt]: spelling: resendEmailC*oun*t

File: app/src/main/java/org/kinecosystem/kinit/view/backup/BackupWalletActivity.kt [hunk 2]
```diff
@@ -10,11 +10,22 @@ import org.kinecosystem.kinit.viewmodel.backup.BackupModel
...
+    private val RESEND_EMAIL_WARNING = 3
 
     override fun getBackUpModel(): BackupModel {
         return model
     }
 
+    override fun onResendClickShouldShowAlert(): Boolean {
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/view/backup/BackupWalletActivity.kt]: logic should reside in model

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    private fun getAlertType(): BackupAlertType {
+        val type = getTypeForCurrentTime()
+        if (!seen(type)) {
+            return type
+        }
+        return BackupAlertType.None
+    }
+
+    private fun getTypeForCurrentTime(): BackupAlertManager.BackupAlertType {
+        return BackupAlertType.BackupNag0
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt]: always the same type ?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private fun seen(type: BackupAlertType): Boolean {
+        return false
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt]: always false?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private fun showPositiveBackupNag(@StringRes title: Int, @StringRes message: Int) {
+        AlertManager.showPositiveAlert(context, R.string.backup_now, R.string.backup_info, R.string.back, {
+            navigator.navigateTo(Navigator.Destination.WALLET_BACKUP)
+        })
+    }
+}
+
+class AlertProperties() {
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/backup/BackupAlertManager.kt]: Is this used?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt [hunk 6]
```diff
@@ -40,7 +42,12 @@ class InfoViewModel(val navigator: Navigator) : TabViewModel {
     }
 
     fun onStartBackupClicked(view: View) {
-        navigator.navigateTo(Navigator.Destination.WALLET_BACKUP)
+        if(isBackedUp.get()){
+            showCreateNewBackupAlert.set(true)
+            showCreateNewBackupAlert.set(false)
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt]: that doesn't look right. kind of hacking around the usage of binding properties... 
I would prefer to add the ability of displaying the popup to the navigator

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt [hunk 7]
```diff
@@ -40,7 +42,12 @@ class InfoViewModel(val navigator: Navigator) : TabViewModel {
...
+        }else {
+            navigator.navigateTo(Navigator.Destination.WALLET_BACKUP)
+        }
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt]: format...

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt (hunk 6 above)
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt]: How about something like:
navigator.navigateUponConfirm(Navigator.Destination.WALLET_BACKUP, AlertToDisplay)

Comment (shayb) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/info/InfoViewModel.kt]: not want to use the context in the model and nagivator is not alert dispalyer
but i solved it in another way

File: app/src/main/java/org/kinecosystem/kinit/view/TabsAdapter.kt [hunk 8]
```diff
@@ -120,8 +126,20 @@ class TabsAdapter :
 
     private fun getInfoTab(parent: ViewGroup, position: Int): View {
         val binding = DataBindingUtil.inflate<InfoTabBinding>(LayoutInflater.from(parent.context),
-            R.layout.info_tab, parent, false)
-        binding.model = InfoViewModel(Navigator(parent.context))
+                R.layout.info_tab, parent, false)
+        val navigator = Navigator(context)
+        var model = InfoViewModel(navigator)
+        model.showCreateNewBackupAlert.addOnPropertyChangedCallback(object : Observable.OnPropertyChangedCallback() {
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/view/TabsAdapter.kt]: you should be able to remove the callback

**Pull request merged**